includes/blocks/course-theme/class-next-lesson.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * File containing the Next_Lesson class.
+ *
+ * @package sensei
+ * @since
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+
+/**
+ * Class Next_Lesson is responsible for rendering the 'Next Lesson >' blocks.
+ */
+class Next_Lesson {
+
+	/**
+	 * The right chevron icon.
+	 *
+	 * @var string
+	 */
+	public static $icon = '<svg viewBox="0 0 8 13" fill="none" xmlns="http://www.w3.org/2000/svg">
+							<path d="M0.999999 1.00002L6 6.50002L1 12" stroke="#1E1E1E" stroke-width="1.5"/>
+						   </svg>';
+
+	/**
+	 * Next_Lesson constructor.
+	 */
+	public function __construct() {
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-next-lesson',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @param array $attributes The block attributes.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render( array $attributes = [] ) : string {
+		$lesson = get_post();
+
+		if ( empty( $lesson ) ) {
+			return '';
+		}
+
+		$urls = sensei_get_prev_next_lessons( $lesson->ID );
+
+		if ( empty( $urls['next']['url'] ) ) {
+			return '';
+		}
+		$url  = esc_url( $urls['next']['url'] );
+		$text = $attributes['text'] ?? __( 'Next Lesson', 'sensei-lms' );
+		$text = wp_kses_post( $text );
+		$icon = self::$icon;
+
+		return ( "
+			<a class='sensei-course-theme-prev-next-lesson-a sensei-course-theme-prev-next-lesson-a__next' href='{$url}'>
```

Comment:
On the blocks JS, I think we came out of that discussion with the idea to not do the work speculatively, since it'll likely be different when we really implement them. 

For markup, the approach we use now for buttons is a bit hacky, and I think they were mostly done because some of the core Wordpress customizations were not readable from block attributes, just in the generated HTML. Even if we end up continuing that practice, right now it'd just add extra work (including maintenance when the markup changes and we need to update the template) without much benefit.

Reply:
> On the blocks JS, I think we came out of that discussion with the idea to not do the work speculatively, since it'll likely be different when we really implement them.

You mean the blocks being different in the future? In terms of HTML, I think it wouldn't be very speculatively because we already know what we want to show to the user. But feel free to disagree. 🙂 The settings I agree that could be a little more speculatively in terms of what we allow to customize in the future.

> For markup, the approach we use now for buttons is a bit hacky, and I think they were mostly done because some of the core Wordpress customizations were not readable from block attributes, just in the generated HTML.

I think for this specific case, since it involves server-side logic to get the URL for the next lesson, we wouldn't be able to do it in a different way than something dynamic `render_callback`. Or is there some way you can think of?

> Even if we end up continuing that practice, right now it'd just add extra work (including maintenance when the markup changes and we need to update the template) without much benefit.

My main point is that if we use `render_callback` to render everything, and the blocks are totally different later, maybe it wouldn't be worth it to think of everything as blocks now because anyway we would need to rewrite most parts. At least having correctly what we serve as HTML and what is dynamic, we would need to just write the missing JS part in the future. Does it make sense?

Reply:
I'm arguing for using dynamic blocks only with a PHP `render_callback`, like it's done in this PR. Everything else (JS components for the block editor, generated HTML block markup) is unnecessary for the project. 

I'd also say that in the future when we do create the JS components for the block editor, it'll still be better to not have them generate any HTML markup in their `save` function, just store everything in attributes and generate the HTML in the render callback. The only benefit of a hybrid approach would be to share some code between the edit and save functions of the block, but in our past work that turned out to be not really happening.

Reply:
I am very new to blocks. I actually started implementing this PR with JS parts and learned a lot about how to create blocks in WP. 😄 Then I threw all that away because the PR started to get too big.

With my limited knowledge, I see @renatho's suggestion more appealing.

My reasoning is, no matter what path we choose we will probably end up rewriting stuff. Because it's a new  feature and not every detail is ironed up yet. However, if we write these blocks as fully featured WP blocks, we will have more context. We will be keeping in mind all the parts of the block while building it and considering potentially more edge cases and such, which might be harder to overcome if we discover them later than sooner.

Another benefit I see for the full blocks approach, is that it is much clearer direction. The direction is "We build blocks and then we use them in our custom course theme templates". The other approach is "We build half blocks with parts that we will be using now, but also make sure the parts that we need in the future is easy to add later." The latter approach is less clear way of doing things. Or maybe it's less clear for me, because I'm new to blocks. 🤔 

That being said, the approach with full blocks require more time. But I think it's worth it.

Reply:
I think if we don't create the JS files it wouldn't be a big deal. The benefit of creating a small version of them is that we could add them temporarily to a lesson and copy the code to the template, so we avoid issues with class names, and so on. But if we decide to not create them for now and trust that we'll create them correctly, that's okay for me! However, I agree with @dadish thoughts that it would be more clear and probably less error-prone.

About the hybrid approach, I think only the outline and the contact teacher block would use more dynamic parts from the server. For the other ones, we would get only small pieces of data from the server, so I think the best practice would probably be the hybrid approach there.